lib/compress/zstd_fast.c

```diff
@@ -73,26 +98,62 @@ size_t ZSTD_compressBlock_fast_generic(
         U32 const current = (U32)(ip-base);
         U32 const matchIndex = hashTable[h];
         const BYTE* match = base + matchIndex;
+        const U32 repIndex = current + 1 - offset_1;
+        const BYTE* repBase = (dictMode == ZSTD_dictMatchState
+                            && repIndex < prefixLowestIndex) ?
+                               dictBase - dictIndexDelta : base;
```

Comment:
minor: 
`dictBase - dictIndexDelta` is supposed to be UB (it's an address outside of the object).

Since `repBase` will never be dereferenced, it's unlikely to degenerate into any kind of problem, therefore if there was some kind of benefit to this (speed typically), this could be debatable.
However, in this case, `repBase` is only used once, so I don't see any benefit invoking UB behavior.

On the other hand, `repMatch = repBase + (repIndex - dictIndexDelta)` is always correct (provided `repIndex >= dictIndexDelta`, which is supposed to be the case, and can be asserted).